pioinstaller/pack/packer.py

```
@@ -40,13 +45,18 @@ def pack(target):
     create_wheels(os.path.dirname(util.get_source_dir()), tmp_dir)
 
     new_data = io.BytesIO()
-    for whl in os.listdir(tmp_dir):
-        with zipfile.ZipFile(os.path.join(tmp_dir, whl)) as existing_zip:
+    for filename in os.listdir(tmp_dir):
+        if not filename.endswith(".whl"):
+            continue
+        filepath = os.path.join(tmp_dir, filename)
+        with zipfile.ZipFile(filepath) as existing_zip:
             with zipfile.ZipFile(new_data, mode="a") as new_zip:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)